exec/java-exec/src/test/java/org/apache/drill/exec/physical/impl/writer/TestParquetWriter.java

```diff
@@ -754,15 +764,45 @@ public void testImpalaParquetVarBinary_DictChange() throws Exception {
     compareParquetReadersColumnar("field_impala_ts", "cp.`parquet/int96_dict_change.parquet`");
   }
 
+  @Test
+  public void testImpalaParquetBinaryTimeStamp_DictChange() throws Exception {
+    try {
+      test("alter session set %s = true", ExecConstants.PARQUET_READER_INT96_AS_TIMESTAMP);
+      compareParquetReadersColumnar("field_impala_ts", "cp.`parquet/int96_dict_change.parquet`");
```

Comment:
This is not a good enough test. Both the baseline and test case queries will use the getDateTimeValueFromBinary method and if there is a bug in that method, the test will still pass as both will produce the same incorrect value. Better to compare with the actual baseline value in the file.